workspace_api/views.py

```diff
@@ -113,6 +127,36 @@ async def create_workspace(
     return {"name": workspace_name}
 
 
+def create_secret(credentials: Dict[str, Any]):
+
+    secret = k8s_client.V1Secret(
+        metadata=k8s_client.V1ObjectMeta(
+            name=CONTAINER_REGISTRY_SECRET_NAME,
+        ),
+        data={
+            "username": base64.b64encode(credentials["workspace_name"].encode()).decode(),
+            "password": base64.b64encode(
+                credentials["password"].encode()
+            ).decode(),
+        },
+    )
+    return secret
+
+
+async def wait_for_secret(
+        credentials: Dict[str, Any], res , time_out: int, time_interval: int):
+
+    timer = 0
+    while res.status == 201:
+        time.sleep(time_interval)
```

Comment:
Using blocking functions in async code is harmful, as it starves the event loop. Use `asyncio.sleep` instead (and `await` it)